app/controllers/concerns/hyrax/valkyrie_downloads_controller_behavior.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+module Hyrax
+  module ValkyrieDownloadsControllerBehavior
+    # before_action :authorize_download!
+
+    def show_valkyrie
+      file_set_id = params.require(:id)
+      file_set = Hyrax.query_service.find_by(id: file_set_id)
+      send_file_contents_valkyrie(file_set)
+    end
+
+    private
+
+    def send_file_contents_valkyrie(file_set)
+      response.headers["Accept-Ranges"] = "bytes"
+      self.status = 200
+      use = params.fetch(:file, :original_file).to_sym
+      file_metadata = find_file_metadata(file_set: file_set, use: use)
+      file = Hyrax.storage_adapter.find_by(id: file_metadata.file_identifier)
+      prepare_file_headers_valkyrie(metadata: file_metadata, file: file)
+      file.rewind
```

Comment:
does this mean the whole file is loaded in to memory at this point? /cc @no-reply